tests/gene-comparison-tool-pinning-cache.spec.ts

```diff
@@ -0,0 +1,433 @@
+
+test.describe('GCT: Caching pinned genes', () => {
+  test('Pinned genes are maintained when switching between RNA subcategories', async ({
+    page,
+  }) => {
+    const genes = ['CYB561A3', 'MANBAL', 'PLEC', 'GFAP'];
+    const nGenes = genes.length;
+
+    await page.goto(URL_GCT);
+    await expectGctPageLoaded(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD
+    );
+
+    await pinMultipleGenesViaSearch(page, genes);
+
+    await test.step('confirm # genes pinned on AD subcategory page', async () => {
+      await expectPinnedGenesCountText(page, nGenes);
+      await confirmPinnedItemsCount(page, nGenes);
+    });
+
+    await changeGctSubcategory(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD,
+      GCT_RNA_SUBCATEGORIES.AD_AOD
+    );
+
+    await changeGctSubcategory(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD_AOD,
+      GCT_RNA_SUBCATEGORIES.AD_SEX_F
+    );
+
+    await changeGctSubcategory(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD_SEX_F,
+      GCT_RNA_SUBCATEGORIES.AD_SEX_M
+    );
+
+    // back to original subcategory
+    await changeGctSubcategory(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD_SEX_M,
+      GCT_RNA_SUBCATEGORIES.AD
+    );
+
+    await test.step('confirm same genes pinned on AD subcategory page', async () => {
+      await expectPinnedGenesCountText(page, nGenes);
+      await confirmPinnedItemsCount(page, nGenes);
+      await confirmPinnedItemsByItemName(page, genes);
+    });
+  });
+
+  test('Pinned genes are maintained when switching categories: RNA -> Protein -> RNA', async ({
+    page,
+  }) => {
+    const genes = ['CYB561A3', 'MANBAL', 'PLEC', 'GFAP'];
+    const nGenes = genes.length;
+
+    await page.goto(URL_GCT);
+    await expectGctPageLoaded(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD
+    );
+
+    await pinMultipleGenesViaSearch(page, genes);
+
+    await test.step('confirm # genes pinned on RNA page', async () => {
+      await expectPinnedGenesCountText(page, nGenes);
+      await confirmPinnedItemsCount(page, nGenes);
+    });
+
+    await changeGctCategory(page, GCT_CATEGORIES.RNA, GCT_CATEGORIES.PROTEIN);
+
+    await changeGctSubcategory(
+      page,
+      GCT_CATEGORIES.PROTEIN,
+      GCT_PROTEIN_SUBCATEGORIES.SRM,
+      GCT_PROTEIN_SUBCATEGORIES.TMT
+    );
+
+    await changeGctSubcategory(
+      page,
+      GCT_CATEGORIES.PROTEIN,
+      GCT_PROTEIN_SUBCATEGORIES.TMT,
+      GCT_PROTEIN_SUBCATEGORIES.LFQ
+    );
+
+    // back to original category
+    await changeGctCategory(page, GCT_CATEGORIES.PROTEIN, GCT_CATEGORIES.RNA);
+
+    await test.step('confirm same genes pinned on RNA page', async () => {
+      await expectPinnedGenesCountText(page, nGenes);
+      await confirmPinnedItemsCount(page, nGenes);
+      await confirmPinnedItemsByItemName(page, genes);
+    });
+  });
+
+  test.fail(
```

Comment:
`test.fail` used to mark tests that will fail until the following changes are made to pinning / caching -- 
1. User is limited to 50 pinned *genes*, not 50 pinned items. For example, 50+ proteins can be pinned for 50 unique genes. 
2. Gene count is displayed separately from protein count on Protein category pages, e.g. "Pinned Genes (50/50)" and "Pinned Proteins (54)".
3. All proteins mapping to a gene are pinned when switching between views, e.g. all 5 GFAP proteins, rather than just the first GFAP protein.